custom/aaa/README.md

```diff
@@ -0,0 +1,96 @@
+queries in the ORM.
+
+# Collection of Data
+
+Collecting data from the application should be done via UCR data sources.
+These are stored in ./ucr/data_sources.
+When writing these care should be taken to ensure there are no (or few) extra queries.
+Each data source only pulls information from directly from the case of form and an id to relate to another data source.
+
+For example, the eligible_couple_forms data source listens for new submissions to the Eligible Couple Form.
+We pull the person case id from the case update, so that we can join it to the relevant person case id during aggregation.
+
+For properties that will (or should) stay the same during the lifetime of the case, we should pull these using a case data source.
+Some examples of this are "dob", "sex", or "name".
+It's technically possible for these to change in the app, but only in the event that the original value was incorrect.
+
+For properties that can change throughout the lifetime of the case, these properties should be pulled from case updates inside forms.
+The reason for this is because we do not store case history in any models.
+An example of this is to determine if a woman is pregnant, we cannot use the "is_pregnant" case property.
+We must use "opened_on" and "add" on the ccs_record to construct the times that the woman has been pregnant (and registered in our system) over her lifetime.
+
+# Aggregation of Data
+
+In general the workflow of our data model is:
+
+UCR(s) --> Beneficiary Model(s) --> Location Aggregate Table
+
+For example to generate the Child table we aggregate data from the following UCRs:
+
+* reach-child_health_cases
+* reach-person_cases
+* reach-household_cases
+* reach-awc_location
+* reach-village_location
+
+
+And to generate our AggAwc table we pull from the Child, Woman and CcsRecord models.
+
+To profile the aggregation queries you can use `custom.aaa.utils.explain_aggregation_queries`
+
+# Scaling Considerations
+
+Currently our aggregations are not going to be performant at a full national scale.
+They must go through our entire dataset to regenerate the beneficiary tables.
+Eventually this aggregation workflow will need to change.
+The most likely option here is to add support for sequential IDs in our UCR framework.
+After this we will use an aggregation workflow as described by the
+[Citus blog](https://www.citusdata.com/blog/2018/06/14/scalable-incremental-data-aggregation/).
+
+Some more discussion on this point can be found in the original [GitHub PR](https://github.com/dimagi/commcare-hq/pull/23243).
+
+There will also be a move to using the Citus postgres extension and distributing these tables over multiple shards.
+With this it will mean that we will need a sharding key and as such we may need related document lookups in data sources to allow them to shard efficiently.
+
+# Testing
+
+To test the aggregation logic we have a set of CSV files in ./tests/data/ucr_tables.
```

Comment:
This stuff is kind of a lie until I figure out what's wrong with https://github.com/dimagi/commcare-hq/pull/23351